### PR TITLE
docs: docs update and overhaul

### LINKED ISSUE #433: [Docs] overhaul and update all docs and diagrams
- [ ] Reviewing all existing docs
- [ ] Updating docs where required
- [ ] Updating diagrams where required
- [ ] Removing developer docs to consolidate maintenance
- [ ] Adding `Supported scenarios` top of the docs for visibility

### PR DESCRIPTION
**Pull request by anfibiacreativa**
- [x] Update all diagrams to February 2024 version of the app and architecture
- [x] Add additional scenarios documentation - IN PROGRESS
- [x] Cleans up and removed outdated docs in Developer Guidelines - IN PROGRESS
When completed closes #433

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -314,11 +263,9 @@ When you are done, you can delete all the Azure resources created with this temp
 azd down
 ```
 
-## AI chatbot integration
+## Developer guidelines
 
-The Contoso Real Estate application can be integrated with an AI support chatbot built using [Azure OpenAI](https://azure.microsoft.com/products/ai-services/openai-service).
-
-For more details about enabling the chatbot integration, see [this section](./docs/chatbot-integration.md).
+The project has a standalone [Developer Guide](./packages/docs/website/README.md) defined under `packages/docs` and implemented as an interactive website using the [Docusaurus](https://docusaurus.io) platform. To learn more about it, [go here](./packages/docs/website/developer-guidelines.md).
```
Comment (DanWahlin) [inline README.md:268]: Good addition. We'll try to get this moved into Training modules in the near future so the content is on a 1P resource and more discoverable.

File: assets/scenarios/scenario1.png
Comment (DanWahlin) [inline assets/scenarios/scenario1.png:1]: Having these broken out scenario diagrams is great! Something we can use elsewhere. I mentioned to Chris that I'd like to create a high-level scenario diagram that simply lists them and what services they use (no an architecture diagram). Then, in each scenario doc we show that overview but highlight the scenario we're covering. Under that we then show this type of architecture image. That way people understand the big picture, where they currently are (which scenario), and see how the services interact through the architecture diagram like this one.

File: docs/README.md [hunk 2]
```diff
@@ -8,7 +8,32 @@ ms.topic: overview #Required
...
 
-# Contoso rentals
+# Enterprise-grade Reference Architecture for JavaScript
+
+
+This repository contains the reference architecture and components for building enterprise-grade modern composable frontends (or micro-frontends) and cloud-native applications. It is a collection of best practices, architecture patterns, and functional components that can be used to build and deploy modern JavaScript applications to Azure.
+
+[Additional details about what the Contoso Real Estate app is and what it does from a high-level – possibly has some images from the running app].
+
+### Supported App Scenarios
```
Comment (DanWahlin) [inline docs/README.md:18]: This is really nice to have for devs first viewing the repo. I see 8 scenarios in the diagram and scenarios docs (https://30daysof.github.io/contoso-real-estate/intro/) though but only 5 here. Do we want this overview to match up with the scenario docs?

File: docs/README.md [hunk 3]
```diff
@@ -24,7 +49,7 @@ The Contoso HR App is part of the Contoso platform and designed to serve interna
 
 ## App architecture
 
-To support this app functionality, the Contoso engineering team decided to build the app with the following:
+To support this app functionality, the Contoso engineering team decided to build the app with the following structure:
```
Comment (DanWahlin) [inline docs/README.md:52]: I'll admit this is a "picky" point to make, but I'd word it with more confidence by removing "decided". Something like:

To support this app functionality, the Contoso engineering team used the following structure:

**Review by DanWahlin — DISMISSED** [review]
@manekinekko and others involved with the project from the start will need to look at some of the specifics in this PR more closely, but from a higher-level I think this provides a solid update to help people understand the project more and get them to resources.

File: docs/README.md (hunk 2 above)
Comment (manekinekko) [reply docs/README.md:18]: FYI, scenario 7 (Redis Caching) has been deprioritized and hasn't been implemented yet.

File: docs/real-time-notifications-scenario.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+## Steps to enable the service
+
+1 - Create a Web PubSub For Socket.IO resource.
+Click key tab in resource portal, copy connection string.
+
+2 - Rename .env.example under packages/realtime/ to .env. Then fill in copied connection string to variable WebPubSubConnectionString
+
+3 - Copy the endpoint part in connection string (e.g. https://<resource-name>.webpubsub.azure.com). And use it replace the default endpoint in packages/portal/src/app/shared/realtime.service.ts:L14
+Follow original README to go on.
```
Comment (manekinekko) [inline docs/real-time-notifications-scenario.md:20]: FYI, these manual steps are temporarily required untill we add Bicep/azd support for Web PubSub.

**Review by manekinekko — DISMISSED** [review]

**Review by manekinekko — APPROVED** [review]

**Pull request merged**